common/src/chunk_loader.rs

```diff
@@ -39,8 +39,32 @@ impl ChunkLoader {
         }
     }
 
+    pub fn load_chunks<'a>(
+        &mut self,
+        graph: &mut DualGraph,
+        dimension: u8,
+        nodes: impl Iterator<Item = &'a NodeId>,
+    ) {
+        for &node in nodes {
+            for chunk in Vertex::iter() {
+                if let Chunk::Fresh = graph
+                    .get(node)
+                    .as_ref()
+                    .expect("all nodes must be populated before loading their chunks")
+                    .chunks[chunk]
+                {
+                    if let Some(params) = ChunkParams::new(dimension, graph, node, chunk) {
+                        if self.load(node, params) {
+                            graph.get_mut(node).as_mut().unwrap().chunks[chunk] = Chunk::Generating;
+                        }
```

Comment:
Pre-existing issue (and hence best fixed in a separate commit, and not necessarily in this PR), but we should immediately `return` if there's no capacity available.